src/engine/routing_algorithms/alternative_path_mld.cpp

```diff
@@ -147,15 +147,19 @@ double scaledAtMostLongerByFactorBasedOnDuration(EdgeWeight duration)
 // Filters candidates with much higher weight than the primary route. Mutates range in-place.
 // Returns an iterator to the filtered range's new end.
 template <typename RandIt>
-RandIt filterViaCandidatesByStretchHeuristic(RandIt first, RandIt last, EdgeWeight weight)
+RandIt filterViaCandidatesByStretchHeuristic(RandIt first,
+                                             RandIt last,
+                                             EdgeWeight weight,
+                                             double weight_multiplier)
 {
     util::static_assert_iter_category<RandIt, std::random_access_iterator_tag>();
     util::static_assert_iter_value<RandIt, WeightedViaNode>();
 
     // Assumes weight roughly corresponds to duration-ish. If this is not the case e.g.
     // because users are setting weight to be distance in the profiles, then we might
     // either generate more candidates than we have to or not enough. But is okay.
-    const auto scaledAtMostLongerBy = scaledAtMostLongerByFactorBasedOnDuration(weight);
+    const auto scaledAtMostLongerBy =
+        scaledAtMostLongerByFactorBasedOnDuration(weight / weight_multiplier);
```

Comment:
@oxidase do you think this is good enough for now until we have durations in the overlay?